### ISSUE TITLE
SmoothQuant with OPT w8a8 - no act_scales from SmoothQuant & why is it needed?

### ISSUE THREAD

**Issue by chenho74** [NVIDIA/FasterTransformer#404]
### Branch/Tag/Commit

N/A

### Docker Image Version

N/A

### GPU name

n/a

### CUDA Driver

n/a

### Reproduced Steps

```shell
https://github.com/NVIDIA/FasterTransformer/blob/release/v5.2_bug_fix_tag/docs/gpt_guide.md#run-meta-opt
```

**Comment by byshiue** [comment]
SmoothQuant is a quantization workflow. The quantization scales can be static or dynamic. 

Now, FT only supports static with pert Tensor scale.

**Comment by chenho74** [comment]
I see, thanks! How do you generate act scales from SmoothQuant that huggingface_opt_convert needed then? Based on the information I entailed above, there seems no clear way?

**Comment by byshiue** [comment]
`huggingface_opt_convert` uses `lambada` as input to get the scales of activation. And the scales of weights are determined by weights directly.

**Comment by chenho74** [comment]
Exactly! So there is no need to give act_scales to `huggingface_opt_convert`, right? Why this in the gpt guide then?

> In this section, the guide mentioned "You must also generate activation records for calibrating the scaling factors." and this command:
python3 examples/pytorch/gpt/utils/huggingface_opt_convert.py -i ../smoothquant/opt-1.3b-smooth/ -o ../nlp-models/ft/test/opt-1.3b-int8/ -i_g 1 -act_scale ../smoothquant/opt-1.3b-smooth.scales.pt

**Comment by byshiue** [comment]
That's used for QAT. If you have a model finetuned by QAT, you can pass the checkpoint and this converter does not run calibration. Otherwise (`-quantize` case), it will run calibration by lambada set to collect the scales.

**Comment by JianchaoTan** [comment]
> That's used for QAT. If you have a model finetuned by QAT, you can pass the checkpoint and this converter does not run calibration. Otherwise (`-quantize` case), it will run calibration by lambada set to collect the scales.

Suppose you do not provide -act_scale ../smoothquant/opt-1.3b-smooth.scales.pt for examples/pytorch/gpt/utils/huggingface_opt_convert.py, then it still will work for converting the model to FT,  but when running the bin/multi_gpu_gpt_example with int8_mode =2, it will report an error about miss loading some weights files. 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.attention.query_key_value.weight.int8.0.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.attention.query_key_value.scale.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.attention.query_key_value.scale_inter.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.attention.query_key_value.scale_out.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.attention.dense.weight.int8.0.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.attention.dense.scale.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.attention.dense.scale_inter.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.attention.dense.scale_out.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.mlp.dense_h_to_4h.weight.int8.0.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.mlp.dense_h_to_4h.scale.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.mlp.dense_h_to_4h.scale_inter.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.mlp.dense_h_to_4h.scale_out.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.mlp.dense_4h_to_h.weight.int8.0.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.mlp.dense_4h_to_h.scale.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.mlp.dense_4h_to_h.scale_inter.bin cannot be opened, loading model fails! 

[FT][WARNING] file ../opt-1.3b-smoothquant-int8/1-gpu/model.layers.23.mlp.dense_4h_to_h.scale_out.bin cannot be opened, loading model fails! 


The logic of "examples/pytorch/gpt/utils/huggingface_opt_convert.py" only saves these files when act_scale is not None.  So can you provide such act_scale file for this demo example? Thanks.
